specification/ml/_types/TrainedModel.ts

```diff
@@ -31,39 +31,114 @@ import {
+   */
+  cache_miss_count: integer
   /** The number of failures when using the model for inference. */
-  failure_count: long
-  /** The total number of times the model has been called for inference. This is across all inference contexts, including all pipelines. */
-  inference_count: long
+  failure_count: integer
   /**
-   * The number of times the model was loaded for inference and was not retrieved from the cache. If this number is close to the inference_count, then the cache is not being appropriately used. This can be solved by increasing the cache size or its time-to-live (TTL). See General machine learning settings for the appropriate settings.
-   * @doc_url https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-settings.html#general-ml-settings
+   * The total number of times the model has been called for inference.
+   * This is across all inference contexts, including all pipelines.
    */
-  cache_miss_count: long
+  inference_count: integer
   /** The number of inference calls where all the training features for the model were missing. */
-  missing_all_fields_count: long
+  missing_all_fields_count: integer
   /** The time when the statistics were last updated. */
   timestamp: Time
 }
 
+export class TrainedModelSizeStats {
+  /** The size of the model in bytes. */
+  model_size_bytes: ByteSize
+  /** The amount of memory required to load the model in bytes. */
+  required_native_memory_bytes: integer
+}
+
+export class TrainedModelDeploymentNodesStats {
+  /** The average time for each inference call to complete on this node. */
+  average_inference_time_ms: double
+  /** The number of errors when evaluating the trained model. */
+  error_count: integer
+  /** The total number of inference calls made against this node for this model. */
+  inference_count: integer
+  /**
+   * The number of threads used by the inference process. This value is limited
+   * by the number of hardware threads on the node; it might therefore differ
+   * from the `inference_threads` value in the start trained model deployment API.
+   */
+  inference_threads: integer
+  /** The epoch time stamp of the last inference call for the model on this node. */
+  last_access: long
+  /**
+   * The number of threads used when sending inference requests to the model.
+   * This value is limited by the number of hardware threads on the node; it
+   * might therefore differ from the `model_threads` value in the start trained
+   * model deployment API.
+   */
+  model_threads: integer
+  /** Information pertaining to the node. */
+  node: DiscoveryNode
+  /** The reason for the current state. Usually only populated when the `routing_state` is `failed`. */
+  reason: string
+  /** The number of inference requests that were not processed because the queue was full. */
+  rejection_execution_count: integer
+  /** The current routing state and reason for the current routing state for this allocation. */
+  routing_state: RoutingState
```

Comment:
`reason` and `routing_state` are both members of the same nested object. 

There is already a class `TrainedModelAllocationRoutingTable` with the correct fields (reason & routing_state) Delete the reason field above and change this to:

```suggestion
  routing_state: TrainedModelAllocationRoutingTable
```

Reply:
Does this mean the layout in https://elasticsearch_82705.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-trained-models-stats.html is wrong, since it shows "reason" and "routing_state" as peer properties in "nodes"?

Reply:
Yes the layout is wrong i've also fixed that in https://github.com/elastic/elasticsearch/pull/82705/commits/e32ab2a54f3ce0922ddc79f5672dfe8d13d4754e

>  It also seems strange to me that routing_state = routing_table in the get stats API but routing_table contains routing_state in the start deployment response.

Yes it's not really a table is it. I'm also a little confused about the `routing_state` nested in a `routing_state`

```
            "routing_state": {
                "routing_state": "started",
                "reason": ""
            }
        },
```

Reply:
Thanks!  I made this change in 2c1f18dd467c440696b48421e5e2e595b74e8c33